daemon/containerd/service.go

```diff
@@ -99,12 +104,53 @@ func (i *ImageService) ReleaseLayer(rwlayer layer.RWLayer) error {
 // LayerDiskUsage returns the number of bytes used by layer stores
 // called from disk_usage.go
 func (i *ImageService) LayerDiskUsage(ctx context.Context) (int64, error) {
-	panic("not implemented")
+	ch := i.usage.DoChan("LayerDiskUsage", func() (interface{}, error) {
+		var allLayersSize int64
+		snapshotter := i.client.SnapshotService(i.snapshotter)
+		snapshotter.Walk(ctx, func(ctx context.Context, info snapshots.Info) error {
+			usage, err := snapshotter.Usage(ctx, info.Name)
+			if err != nil {
+				return err
+			}
+			allLayersSize += usage.Size
+			return nil
+		})
+		return allLayersSize, nil
+	})
+	select {
+	case <-ctx.Done():
+		return 0, ctx.Err()
+	case res := <-ch:
+		if res.Err != nil {
+			return 0, res.Err
+		}
+		return res.Val.(int64), nil
+	}
 }
 
 // ImageDiskUsage returns information about image data disk usage.
 func (i *ImageService) ImageDiskUsage(ctx context.Context) ([]*types.ImageSummary, error) {
-	panic("not implemented")
+	ch := i.usage.DoChan("ImageDiskUsage", func() (interface{}, error) {
+		// Get all top images with extra attributes
+		images, err := i.Images(ctx, types.ImageListOptions{
+			Filters:        filters.NewArgs(),
+			SharedSize:     true,
+			ContainerCount: true,
+		})
+		if err != nil {
+			return nil, fmt.Errorf("failed to retrieve image list: %v", err)
+		}
+		return images, nil
```

Comment:
It looks like the existing implementation for graph drivers is exactly the same, and also has the same issue, but I noticed two things here;

- can you rename the `images` variable (as it collides with the `images` import)
- the `fmt.Errorf()` is masking the underlying error-type, which would be a `errdefs.InvalidParam`, so we should import `pkg/errors` and use `errors.Wrap()` here.


```suggestion
		imgs, err := i.Images(ctx, types.ImageListOptions{
			Filters:        filters.NewArgs(),
			SharedSize:     true,
			ContainerCount: true,
		})
		if err != nil {
			return nil, errors.Wrap(err, "failed to retrieve image list")
		}
		return imgs, nil
```

Reply:
Looks like the `images` variable colliding is not an _actual_ issue in the graphdriver implementation, but perhaps it's good to make the same changes in that code as well (as a separate commit) to keep the code the same.